packages/mobile/src/account/saga.ts

```diff
@@ -156,6 +166,58 @@ export function* watchKycStatus() {
   }
 }
 
+export function* generateSignedMessage() {
+  try {
+    const address = yield select(currentAccountSelector)
+    const mnemonic = yield call(getStoredMnemonic, address)
+    const { privateKey } = yield call(
+      generateKeys,
+      mnemonic,
+      undefined,
+      undefined,
+      undefined,
+      bip39
+    )
+
+    const signedMessage = yield call(
+      serializeSignature,
+      signMessage('valora auth message', ensureLeading0x(privateKey), address)
+    )
+    yield put(saveSignedMessage(signedMessage))
+
+    return signedMessage
+  } catch (error) {
+    throw error
+  }
+}
+
+export function* handleUpdateAccountRegistration(properties: RegistrationProperties) {
+  const address = yield select(currentAccountSelector)
+  let signature = yield select(signedMessageSelector)
+
+  try {
+    if (!signature) {
+      signature = yield call(generateSignedMessage)
```

Comment:
Just wanted to suggest an alternative, since the user might just refuse to put in their PIN and we'd never get the signed message if they do so:

We can hook into the `getConnectedUnlockedAccount` saga and _then_ create and store the signed message. That way we can make sure that the user has a signed message if they invite someone since that requires unlocking the account. 

We could also create the signed message when creating or importing the account for new users.